src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -84,7 +84,15 @@ namespace
         }
         void operator() (const UUID & x) const
         {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+            auto tmp_x = x.toUnderType();
+            char * start = reinterpret_cast<char *>(&tmp_x);
+            char * end = start + sizeof(tmp_x);
+            std::reverse(start, end);
+            operator()(tmp_x);
```

Comment:
As we call `operator()(const & UInt128)` (l. 57), I would expect that the same endianness issue in principle occurs for all integer types with 128 or more bits (see [here](https://clickhouse.com/docs/en/sql-reference/data-types/int-uint)). Would you like to check that? Perhaps the fix needs to be moved to l. 57/63/75/81.

Reply:
Refactored accordingly.